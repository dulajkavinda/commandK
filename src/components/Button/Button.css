.search-bar {
  display: flex;

  justify-content: center;
  align-items: center;

  border: 2px solid #1e262f;
  background-color: #1e262f;
  color: #e2e2e2;

  width: 11em;
  height: 2.8em;
  padding: 5px 9px;

  border-radius: 8px;

  font-size: 0.8rem;
  cursor: pointer;
}

.search-bar:focus {
  border: 1px dashed #7e8c9a;
}

.search-bar-icon {
  font-size: 1.2em;
  margin-right: 4px;
  font-weight: 500;
}

.search-bar-key {
  color: #7e8c9a;
  border-radius: 3px;
  border: 1px solid #7e8c9a;
  padding: 2px 4px;
  margin-left: 5px;
  font-size: 0.7rem;
}

.search-bar-small {
  width: 9em;
  height: 2.5em;
  padding: 5px 9px;
  font-size: 0.7rem;
}

.search-bar-icon-small {
  color: #e2e2e2;
  font-size: 0.9em;
  margin-right: 4px;
  font-weight: 500;
}

.search-bar-key-small {
  color: #7e8c9a;
  border-radius: 3px;
  border: 1px solid #7e8c9a;
  padding: 2px 4px;
  margin-left: 1px;
  font-size: 0.5rem;
}

.search-bar-large {
  width: 14em;
  height: 3.5em;
  padding: 5px 9px;
  font-size: 0.8rem;
}

.search-bar-icon-large {
  color: #e2e2e2;
  font-size: 1.5em;
  margin-right: 4px;
  font-weight: 500;
}

.search-bar-key-large {
  color: #7e8c9a;
  border-radius: 3px;
  border: 1px solid #7e8c9a;
  padding: 2px 6px;
  margin-left: 5px;
  font-size: 1rem;
}
