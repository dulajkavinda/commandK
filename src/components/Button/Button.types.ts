export type ButtonSize = 'small' | 'medium' | 'large'

export interface ButtonType {
  keyLetter?: string
  size?: ButtonSize
  styles?: React.CSSProperties
  onClick?: () => void
}
