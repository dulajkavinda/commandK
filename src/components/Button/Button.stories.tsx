import type { Meta, StoryObj } from '@storybook/react'
import Button from './Button'

const meta = {
  title: 'Button',
  component: Button,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {},
} satisfies Meta<typeof Button>

export default meta

type Story = StoryObj<typeof meta>

export const Small: Story = {
  args: {
    keyLetter: 'K',
    size: 'small',
  },
}

export const Medium: Story = {
  args: {
    keyLetter: 'B',
    size: 'medium',
  },
}

export const Large: Story = {
  args: {
    keyLetter: 'K',
    size: 'large',
  },
}

export const Custom: Story = {
  args: {
    keyLetter: 'B',
    size: 'medium',
    styles: {
      backgroundColor: 'white',
      color: 'black',
    },
  },
}
