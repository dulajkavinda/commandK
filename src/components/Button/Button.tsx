import React from 'react'
import classNames from 'classnames'
import './Button.css'
import { ButtonSize, ButtonType } from './Button.types'

const Button = (props: ButtonType) => {
  const btnClass = classNames('search-bar', {
    [`search-bar-${props.size}`]: props.size,
  })

  const iconClass = classNames('search-bar-icon', {
    [`search-bar-icon-${props.size}`]: props.size,
  })

  const keyClass = classNames('search-bar-key', {
    [`search-bar-key-${props.size}`]: props.size,
  })

  return (
    <>
      <button
        id='search-button'
        type='button'
        onClick={() => {
          if (props.onClick) {
            props.onClick()
          }
        }}
        className={btnClass}
        style={props.styles}
      >
        <div className={iconClass}>
          <span>Command +</span>
        </div>
        <div className={keyClass}>{props.keyLetter?.toUpperCase()}</div>
      </button>
    </>
  )
}

Button.defaultProps = {
  keyLetter: 'K',
  size: 'medium' as ButtonSize,
}

export default Button
