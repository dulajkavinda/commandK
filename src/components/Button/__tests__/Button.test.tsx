import React from 'react'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Button from '../Button'

describe('components/Button', () => {
  it('should render a button', () => {
    const button = renderer.create(<Button />).toJSON()
    expect(button).toMatchSnapshot()
  })

  it('button should be rendered with a default key as K', () => {
    render(<Button keyLetter='K' />)
    const primaryButton = screen.getByRole('button')

    expect(primaryButton).toHaveClass('search-bar')
    expect(primaryButton).toHaveTextContent('K')
  })

  it('button should be rendered with a given text', () => {
    render(<Button keyLetter='N' />)
    const primaryButton = screen.getByRole('button')

    expect(primaryButton).toHaveClass('search-bar')
    expect(primaryButton).toHaveTextContent('N')
  })

  it('by default button should be rendered with medium size', () => {
    render(<Button />)
    const primaryButton = screen.getByRole('button')

    expect(primaryButton).toHaveClass('search-bar-medium')
  })

  it('button should be rendered with a given size', () => {
    render(<Button size='small' />)
    const primaryButton = screen.getByRole('button')

    expect(primaryButton).toHaveClass('search-bar-small')
  })

  it('button should be clickable', () => {
    const onClick = jest.fn()
    render(<Button onClick={onClick} />)
    const primaryButton = screen.getByRole('button')

    primaryButton.click()
    expect(onClick).toHaveBeenCalled()
  })

  it('button should show given custom styles', () => {
    render(<Button styles={{ backgroundColor: 'red' }} />)
    const primaryButton = screen.getByRole('button')

    expect(primaryButton).toHaveStyle('background-color: red')
  })
})
