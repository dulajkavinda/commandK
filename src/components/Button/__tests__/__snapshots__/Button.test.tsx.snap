// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button should render a button 1`] = `
<button
  className="search-bar search-bar-medium"
  data-testid="search-button"
  id="search-button"
  onClick={[Function]}
  type="button"
>
  <div
    className="search-bar-icon search-bar-icon-medium"
  >
    <span>
      Ctrl +
    </span>
  </div>
  <div
    className="search-bar-key search-bar-key-medium"
  >
    K
  </div>
</button>
`;
